### ISSUE TITLE
✈️ 🚀 Feature Request: Add support for the latest Font Awesome version (Migrated)

### ISSUE THREAD

**Issue by thejaydip** [elementor/elementor#18723]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Upgrade Icon Library From Font Awesome 5 to Font Awesome 6. So we can use the new Thin icon style.

### Describe the solution you'd like

Please Font Awesome library updated to 6

**Comment by frm1001xplrr** [comment]
Seconded

**Comment by oceandiveloper** [comment]
yes please

**Comment by adamjacobi** [comment]
Font Awesome 6 has been out for months. Is this even committed to by Elementor?

**Comment by hitsgee** [comment]
Any news on this? Also adding Googles open source icons would be great.

**Comment by crispee2000** [comment]
Yes.  Can't believe this isn't in.

**Comment by mettrodigital** [comment]
Yes please!

**Comment by jxpx777** [comment]
Looks like there is a stalled PR to upgrade: https://github.com/elementor/elementor/pull/18546/files

**Comment by pan-athen** [comment]
I'm not aware of the underlying tech here, so I'm sorry if what I propose cannot be done. I'm just pitching the high-level idea.

I would like to add that the rest of the icon styles that come out of beta would also be nice to be included. Like the newer "Sharp Solid" which looks beautiful. Here's a preview: https://fontawesome.com/search?s=solid&f=sharp&o=r

In general, Font Awesome seems committed to producing many different styles. The Sharp family is going to get extended soon. See here: https://fontawesome.com/docs/web/dig-deeper/styles

A future-proof solution for Elementor would be to include styles that come out of the beta track of Font Awesome as soon as possible. Can this be made at least for those who use the pro version of FA and kits?

Cheers!

**Comment by EdwardBlack** [comment]
Half a year later.... come on elementor!

**Comment by hitsgee** [comment]
It's ridiculous! Elementor thinks they're cutting-edge, but they are behind the game now.

As a developer, I have missed out on so many opportunities to create incredible designs with the new thin, thick styles etc. Along with having to design a ridiculous amount of Covid-19 info pages during the pandemic without a Covid Icon available due to lack of updates!

- Alpha release date was 27th October 2020 **(2+ Years ago)**
- Beta release date was 29th May 2021 **(18+ Months ago)**
- Official release date was 7th February 2022 **(10+ Months ago)**

As you can see from the dates above, Elementor indeed were aware of FA6 and could easily have implemented the library update during the Alpha/Beta stages and released it to us in the first update after the official FA release.

It's also worth noting FA is now on 6.2.1 with five updates since the official release, so it's got to be stable!

Come on, Elementor, get your act together!

**Comment by pan-athen** [comment]
> As you can see from the dates above, Elementor indeed were aware of FA6 and could easily have implemented the library update during the Alpha/Beta stages and released it to us in the first update after the official FA release.
> 
> It's also worth noting FA is now on 6.2.1 with five updates since the official release, so it's got to be stable!
> 
> Come on, Elementor, get your act together!

Second that. This is mandatory for modern and visually appealing websites. I'm stuck with uploading files with any icon I need and using it via the media gallery, which is something I would gladly do in the 90s.

**Comment by adamjacobi** [comment]
I submitted a support ticket and asked to be escalated to someone who could give a direct answer, and just got this stock response:

Hi Adam,

Thank you for contacting Elementor. I’m Nitesh from Tier 2 Customer Experience.

Font Awesome 6 implementation request has been submitted to the product development team for review already.

Unfortunately, we can’t provide any ETA, I suggest keeping an eye on the Changelog to know when it's released.
https://elementor.com/pro/changelog/

I hope it helps. Please don't hesitate to create a new ticket from your [Account](https://my.elementor.com/) if a new issue arises.

We'd be glad to help.

Have a great day/evening.

Best Regards,

**Comment by lllopo** [comment]
+100

**Comment by teaqueue** [comment]
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ FA6 pls ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

**Comment by poliquinp** [comment]
Adding my name to the request for update! 
I'll stay tuned for the release of the feature.

**Comment by Carl-Wilhelm** [comment]
Well come on Elementor! This is just stupid by now!

**Comment by tobias992** [comment]
Yes, welcome on board. This company is really a joke. Prefer to implement something buggy instead of first taking care of elementary basic functions. Just embarrassing. And for this s*** i pay money.

**Comment by aalandes** [comment]
+1 Having to manually upload FA6 icons is tedious af! Come on Elementor we need this already :|

**Comment by CarterStevens1** [comment]
+1

**Comment by lllopo** [comment]
I was really surprised that they need to do something manually about it at first place. Even if they need some metadata that is not present in the FontAwesome download, Elementor is a company big enough to strike some deal with these guys to receive the necessary info like tags for the search etc, so they can automate the updates with each FA release. Honestly ... I don't get it why the manual work at all ... and the end result - those huuuuuuge delays that are (rightfully) pissing off everyone.

**Comment by adamjacobi** [comment]
I'm beyond frustrated with this company's lack of responsiveness -- they just do not seem to care. I am starting to consider finding a different builder platform for my site and abandon Elementor altogether. Perhaps if we all started leaving negative reviews on Google, the Wordpress plugin site, etc. - relative to Elementor's lack of customer service on this issue, perhaps they would take notice? I've worked with dozens of tech companies, and this one, by far, is the most uncompassionate, particularly with this issue.

**Comment by pan-athen** [comment]
Even with big budgets, companies still have limited resources, and being a company owner myself, I can say that we can never know what makes something possible or impossible when we're on the outside. I don't think that Elementor people are uncompassionate or that they don't care. I'm sure that they understand by now that many people like this feature to be enhanced. I believe that constructive criticism and continuous posting in this thread will gather more focus from the decision-makers within Elementor. So, I propose to share this issue with as many web developers and designers as you know that use Elementor and explain to them the features of Font Awesome and this integration. Let's try to convince by numbers.

**Comment by lphaweb** [comment]
I completely second this... and I hope it will soon be implemented. I've bought fontawesome pro thinking it would be well compatible with Elementor as advertised... what a shame it was to discover there are plenty of pro icons you can't use... even with Elementor Pro.

**Comment by pan-athen** [comment]
> I completely second this... and I hope it will soon be implemented. I've bought fontawesome pro thinking it would be well supported by Elementor, as it was advertised... what a shame it was to discover there are plenty of pro icons you can't use... even with Elementor Pro... because Elementor uses an old (and pretty outdated) icon set.

Yes, and the new styles from Font Awesome are so cool and contemporary.

**Comment by markbosky** [comment]
+1

Also, if FA6 is not supported, there should be a notice next to the FontAwesome Kit ID field in Elementor Settings stating so to avoid futile troubleshooting.

**Comment by adamjacobi** [comment]
This — and the clearly dismissive attitude of the support reps I’ve submitted tickets to on this issue — has left such a bad impression of Elementor, that I’ve been slowly but surely converting all my Elementor pages to Wordpress blocks (Gutenberg), and finding they load SO much FASTER. I lose a *little* flexibility, but blocks have come so far AND I have found one that supports FA 6. I’m looking forward to nonrenewing my Elementor Pro license and saying farewell to a disappointing company.

**Comment by Apfelbiss** [comment]
I'm also waiting for this to get fixed!

**Comment by jarimustonen** [comment]
The current version of Font Awesome is 6.4.0. It's totally inexcusable that Elementor does not support this. I just bought Font Awesome license just to learn that I can't use the icons I want as Elementor does not support this. How is this even possible?

**Comment by tobias992** [comment]
> The current version of Font Awesome is 6.4.0. It's totally inexcusable that Elementor does not support this.

Elementor is the most buggy page builder on market. They only release buggy new features to keep the fanboys happy. In every new release discussion i read "what a great update" - i always start laughing when i read this. The 3.14 should be a bugfix release - no new features or anything else. Here are hundreds of open issues - but elementor is not interested in customer feedback. Only buggy new features are important to make more money. In my eyes, this whole company has now become a big joke.

**Comment by masseym** [comment]
Add my vote here too.

**Comment by vgstef** [comment]
Any ETA about when Font Awesome will be updated?

**Comment by jarimustonen** [comment]
Here's a workaround: Register to Font Awesome as paying customer, download the icons as SVG, and use Elementor Icon's **Upload SVG** option.

Here's the button to download the SVG:

<img width="1166" alt="Screenshot 2023-06-29 at 8 41 00" src="https://github.com/elementor/elementor/assets/1272053/fc059c99-f56b-402d-8a4f-b816469efc31">

And here's the location to upload it:

<img width="299" alt="Screenshot 2023-06-29 at 8 42 31" src="https://github.com/elementor/elementor/assets/1272053/95633174-acfe-4575-bf08-c799c8b95a7e">

I hope this helps.

**Comment by pan-athen** [comment]
> Here's a workaround: Register to Font Awesome as paying customer, download the icons as SVG, and use Elementor Icon's **Upload SVG** option.
> I hope this helps.

I really appreciate the workaround and the time you took to post it, and I've used it a lot.

But as a paying customer of Elemenbtor and FA, I find it really strange that something like this cannot happen, especially in the span of time that people post here.

Since this feature request was posted, we have discovered AI in content creation, Elementor can get written text and produce an image, and hosting services have become half the price and 5x the speed. Font Awesome introduced tools to compose your own icons using icons from the library and include them in your kit.

I really understand the way a company moves with updates and features as I own a software company for 12 years, but this looks like it's an easy task.

Can any of the Elementor people comment on why this is not happening? Where is the problem? It seems that everybody would benefit from this; icons are a main element of visual communication and FA seems to have accomplished more than any other icon development company.

**Comment by SteidleSolutions** [comment]
Well over a year now and still no Font Awesome 6 support. This is insane.

**Comment by JoeHana** [comment]
By the time FA6 gets finally integrated, Font Awesome will already release FA7...

**Comment by adamjacobi** [comment]
This issue was the last straw that cost Elementor my business as a user of their product as well as subscriber to their pro plan. I have fully converted several sites to Wordpress’ block builder, and noticed a considerable speed improvement, as well… not to mention Font Awesome 6 support. What was most telling to me was the dismissive attitude of their tech support and developers when I raised this issue. As a consumer and as someone who runs a business and consults, I am making purchasing choices more and more based on company values and how they are manifest in that company’s employees and attitudes. Farewell, Elementor.

**Comment by classbasics** [comment]
If you go to your Font Awesome 6 Pro portal and create a kit with no icons.
![image](https://github.com/elementor/elementor/assets/3071094/6b4ab21a-a9f7-4519-a6d2-57bd1fd9e8d4)
then you have a Kit with all the FA6 icons.
In Elementor, Settings, Integrations
add your FA pro key and Kit number
![image](https://github.com/elementor/elementor/assets/3071094/51dc82a5-1dbc-4efa-82c5-1aa20d27ab7d)
then Save
![image](https://github.com/elementor/elementor/assets/3071094/2b0c8405-7cc2-451a-a41b-887b9d3220b0)
in Elementor, Icon
![image](https://github.com/elementor/elementor/assets/3071094/89d51255-9290-4dd5-9719-e500569c21b6)
go to Icon Library
![image](https://github.com/elementor/elementor/assets/3071094/6c1b5794-401c-4f91-a239-5677c10ccdd6)

**Comment by pan-athen** [comment]
> If you go to your Font Awesome 6 Pro portal and create a kit with no icons ... then you have a Kit with all the FA6 icons.

No, you don't, where is the:

- Thin
- Sharp/Solid
- Sharp/Regular
- Sharp/Light

???

Even in your screenshot, you can see only a part of the icon families included in the FA Pro tier. That is what people say here, we need a true integration with FA which doesn't exist.

**Comment by classbasics** [comment]
You are right.
I got the information from FontAwesome and thought it worked.
I have emailed them your findings and hope they will fix their Kit setup to include all styles.

**Comment by pan-athen** [comment]
> I have emailed them your findings and hope they will fix their Kit setup to include all styles.

Thanks for reporting back!

It would be really great to see a solution to this finally :-)

**Comment by lllopo** [comment]
I think FA support should be re-worked totally. What "FA {put version here} support" means at all ?!?! FA has an API and Elementor should just use it and work with the latest and greatest available and that's it. I don't know how these guys from Elementor implemented the FA support, but it is definitely not good by design. I won't even start about the lack of any reasonable reaction in this thread and the fact that the (still but not for long as it seems) leading page builder for the leading CMS system refuses to offer proper support for the leading icon library ... it would have been ridiculous if it wasn't so tragic.

**Comment by classbasics** [comment]
FontAwesome said
"Most likely what is happening is that while you have added your Pro Kit to Elementor, with the kit's version set to v6, Elementor's icon chooser doesn't yet know to include icons that are new to v6 among the available options. Last I knew about how Elementor had implemented their Font Awesome integration, they were not using our GraphQL API to determine the available icons for discovery in their icon chooser, for the particular version of Font Awesome that your kit uses. I think they used a snapshot of the icons list from a previous version of Font Awesome. If that's the case, then their list of available icons and styles is simply out of date, and could only be brought up to date by releasing a new version of their plugin. So any icons that were added after their snapshot of the icons list would not show up. And any styles that were added wouldn't show up there either. For example, the Thin style was added in v6, and it probably is not yet available in the Elementor icon chooser.

You still have access to all of the v6 icons and styles with your Kit on your WordPress site, even if they won't show up in Elementor. For example, with your kit installed, you could add an icon by adding the raw HTML tag to a page

If this seems like what's going on, I would recommend letting Elementor know that you'd like to see all icons and styles available through your kit in their icon chooser. That could help them assess the value of improving this feature in their software."

**Comment by lllopo** [comment]
@classbasics Ow ... they know what we all are asking for very well ... they simply don't do it, not even reacting on it.

**Comment by Apfelbiss** [comment]
I also think Elementor itself should know best what to do.

The changelog for version 3.1.0 from 2021-02-13 says:
"Tweak: Upgraded Font Awesome Pro library to v5.15.1"
This should be the current snapshot of the Font Awesome Icon Library.

Either Elementor needs to update this snapshot to support new icons and styles. Or better yet, they add proper support for the Font Awesome API so that you don't always have to wait for an Elementor update after a new versions of Font Awesome.

But after so long, I've given up hope for that. 
Probably not without reason they refrain from a concrete advertisement of Font Awesome support on the Elementor website ...

**Comment by classbasics** [comment]
![image](https://github.com/elementor/elementor/assets/3071094/dbb72455-f1e5-490d-a2d0-992002a0fec4)

**Comment by classbasics** [comment]
Maybe there is something in the new Elementor, but at least you can use the sharp FA6 icons if you need to via HTMl.
![image](https://github.com/elementor/elementor/assets/3071094/a50f007c-b918-4e51-a951-99bc92fcfcd6)

**Comment by Apfelbiss** [comment]
Yeah, RAW HTML is possible, but that doesn't work for all the different Elementor widgets with icon support. The only workaround is to upload the icon as SVG. But then you don't even need support for Font Awesome by Elementor.

**Comment by classbasics** [comment]
Yes you can upload all the SVG to Media.
The problem is the different syles have the same name.
You would have to rename each icon with a relevant style prefix to upload them all.
![image](https://github.com/elementor/elementor/assets/3071094/1536140f-8e0c-4cd4-a478-bf00772d9d1d)

**Comment by classbasics** [comment]
I tried Elementor's DEV version without improvement.
It may be possible to put the FA6 icons into the format of Fontello, IcoMoon or Fontastic as Elementor can upload this.

Another option is flaticons.
https://www.flaticon.com
https://www.youtube.com/watch?v=qYESmNhDxM0
The youtube video also shows a technique that could be used to upload one of the styles from FA6 to the Media folder.
Or put a style appropriate prefix on all the FA6 icons and upload them all to the Media folder.
![image](https://github.com/elementor/elementor/assets/3071094/2ba3a599-9cd6-4438-9f48-0c48f3789899)
Maybe multiple folders in the Media folder would help.
e.g. via a plugin
https://www.joomunited.com/news/how-to-manage-media-with-folders-in-elementor-widgets#:~:text=In%20order%20to%20create%20a,automatically%20create%20the%20new%20subfolder.
I will try these later, but there does seem to be a variety of work-a-rounds.

**Comment by pan-athen** [comment]
It's very kind to see all the solution people are offering using alternative ways, but we really need a good integration with FA so that our FA Kits are recognized and supported directly in Elementor without any update. And this feature request is about that; it's not a question or a how-to.

Of course, uploading the icons as images/SVG or icon sets can work, but this isn't what this feature request is about.

The description of this request should change to include any FA version and for the integration to be live/dynamic. Also, Elementor staff should let us know if there is any chance of this happening or start looking for alternative ways to implement our FA kits. What I don't want to happen is to create icon sets or upload icons in the media library and organize them and use them everywhere like this, and then one day Elementor comes with an update and supports FA Kits right out of the box and I will have to change everything from the start. So, a reply on what the future holds for this feature request from Elementor would be nice, even if it's a negative answer.

**Comment by classbasics** [comment]
![image](https://github.com/elementor/elementor/assets/3071094/dd5c7270-4158-44bb-a675-64d8eb3a321a)

**Comment by classbasics** [comment]
https://elementor.com/help/custom-icons-pro/
was previously discussed. 
It cannot be used to upload the FA6 icons directly downloaded (*.zip from FA6 pro)
You can copy files into Fontello, Icomoon, or Fontastic formats, but it is time-consuming/complicated and gives a similar result as the subfolders in the Media folder.

**Comment by classbasics** [comment]
![image](https://github.com/elementor/elementor/assets/3071094/445d033c-63ef-4827-a3ba-1a8aff925080)

**Comment by lphaweb** [comment]
Has anyone found out where the file containing the icon list and/or that elementor system work? We could try to tweak it on our own instead of waiting for them v_v

**Comment by portrayaloflife** [comment]
+1, would be great to have the new Twitter X icon

**Comment by amirhmoradi** [comment]
Hi, I managed to add Font Awesome 6 Pro (latest) to elementor, with all the inner workings too (i.e. caching, loading only when used, v4 and v5 backward compatibility, added all available styles too, and it also has the latest Twitter X icon ;) ...)

See the screenshot from my elementor's icon library below:
<img width="1195" alt="elementor-icons-font-awesome-pro-6--by-amir-moradi" src="https://github.com/elementor/elementor/assets/1281163/35c81bf5-1f4c-4b22-8356-117510f6874a">

I spent a lot of time and effort to get this working.
If at least 10 people are interested to pay a small fee for this, I will build a plugin for easy use on your website.
=> React to my comment with a thumbs-up or like or smile or any thing positive so I can count you in :).

Plugin price would be 10 EURO, one-time payment.

**Comment by EdwardBlack** [comment]
> I spent a lot of time and effort to get this working. If at least 10 people are interested to pay a small fee for this, I will build a plugin for easy use on your website. => React to my comment with a thumbs-up or like or smile or any thing positive so I can count you in :).
> 
> Plugin price would be 10 EURO, one-time payment.

The only right way in my opinion is to contact Elementor and offer them your code for a one time payment. It's their responsibility!

**Comment by tobias992** [comment]
Exactly. They should pay you 1000€. It's a shame they can't do this feature themselves. But hey... we have "AI" and a lot of other buggy new alpha-status bull****.

**Comment by classbasics** [comment]
10€ is good when compared to the annual price of FA pro.

**Comment by EdwardBlack** [comment]
> 10€ is good when compared to the annual price of FA pro.

You mean on top of the 99$ for Fontawesome Pro and 59$ for Elementor Pro?

**Comment by amirhmoradi** [comment]
@classbasics Thanks for your message/interest. The 10 EURO price is only for my work. it does not include Font Awesome Pro nor Elementor Pro (elementor pro is not required as far as I know...still to be tested once there is enough interest in my work). I will update my previous comment to make this clearer.

**Comment by pan-athen** [comment]
It's an honest effort but I really prefer to see this feature officially supported by Elementor. I have had enough with the 100 extra plugins I have to install just to get Elementor small functionalities that should be there to begin with. They have a tool for icons and implemented Font Awesome in there, they should do the complete job and finish this feature properly. Sell them your code I say! :-) By the way, good luck if you do business and start offering your plugin, just make sure that the 10 EUR you price it covers the future work you will have to do to maintain it for all the updates that Elementor will bring your way ;-) It's not an easy job and half of the older plugins made for Elementor, right now stay unmaintained with only bug reports in their support system.

**Comment by masseym** [comment]
im interested

**Comment by ThatGirlSlays** [comment]
> Hi, I managed to add Font Awesome 6 Pro (latest) to elementor, with all the inner workings too (i.e. caching, loading only when used, v4 and v5 backward compatibility, added all available styles too, and it also has the latest Twitter X icon ;) ...)
> 
> See the screenshot from my elementor's icon library below: <img alt="elementor-icons-font-awesome-pro-6--by-amir-moradi" width="1195" src="https://user-images.githubusercontent.com/1281163/267305090-35c81bf5-1f4c-4b22-8356-117510f6874a.png">
> 
> I spent a lot of time and effort to get this working. If at least 10 people are interested to pay a small fee for this, I will build a plugin for easy use on your website. => React to my comment with a thumbs-up or like or smile or any thing positive so I can count you in :).
> 
> Plugin price would be 10 EURO, one-time payment, for my work (does not include font awesome pro, nor elementor pro (elementor pro is not needed as far as my tests show now... need to be tested when people show interest in my code).

I'm interested

**Comment by chunibyuu** [comment]
Still no update with this ? Come on Elementor !

**Comment by portrayaloflife** [comment]
+1

**Comment by apticknor** [comment]
+1

**Comment by toineenzo** [comment]
https://github.com/elementor/elementor/pull/24315
Something's in the work guys!

**Comment by pan-athen** [comment]
> #24315 Something's in the work guys!

That would be awesome (pun intended) :-) Thanks for the heads up!

**Comment by vHeemstra** [comment]
Hope they finish it before FA 7 is released.. 😂

**Comment by pan-athen** [comment]
> Hope they finish it before FA 7 is released.. 😂

My thoughts exactly! :-P

**Comment by SteidleSolutions** [comment]
> #24315 Something's in the work guys!

About fricking time. I hope it's the latest version of 6 (6.5?) but at this point I'll take what I can get. Any timeline on when it will be pushed?

**Comment by toineenzo** [comment]
> > #24315 Something's in the work guys!
> 
> About fricking time. I hope it's the latest version of 6 (6.5?) but at this point I'll take what I can get. Any timeline on when it will be pushed?

yes, looking at the commits, they're using the latest FA version/

**Comment by toineenzo** [comment]
FYI: FA6 is now added to the roadmap
<img width="302" alt="image" src="https://github.com/elementor/elementor/assets/4921397/81143963-bf96-4367-b5a3-8963321f5816">

**Comment by tobias992** [comment]
Just a joke that they put it on the roadmap. It should be normal to update FA. But okay, next to all the new buggy alpha features, maybe this really is a "big new stable" feature xD

**Comment by toineenzo** [comment]
> Just a joke that they put it on the roadmap. It should be normal to update FA. But okay, next to all the new buggy alpha features, maybe this really is a "big new stable" feature xD

The problem is that the current FA v4 to v5 upgrader was only made to upgrade v4 to v5. Therefore the upgrade mechanism had to be completely rewritten. Looks like (according to https://github.com/elementor/elementor/pull/24315) it is now build in a way that it can also be used for future FA versions like upgrading from v6 to v7. Elementor also needs to make sure it is backwards compatible with people that are still using v4 and that it doesn't break all the set icons on all pages. As you can see the dev working on this issue already has made lots of changes and improvements to his code, so it is a lot of work.

**Comment by tobias992** [comment]
**My opinion** is that elementor cannot be backwards compatible all the time. This breaks the builder and the plugin code is very bloated. Sometimes you have to say "breaking change". So people have to update their websites. If they don't want, don't run the update. This backwards compatibility is the reason why elementor is one of the slowest and most bloated builder on the market. Again, my opinion.

**Comment by vHeemstra** [comment]
> As you can see the dev working on this issue already has made lots of changes and improvements to his code, so it is a lot of work.

Even so, for a software package with these prices and with this many customers, the quality and speed of updates should be much better!

Looking at several issues on this GitHub, there's a lot of buggy stuff that should never have been in the code in the first place and missing features that are requested years ago without answers.

But I must say, there seems to be some improvement on Elementor's side recently.

**Comment by Paul-Morris** [comment]
2 years just to add it to the roadmap? Clients are starting to complain about the old Twitter logo. Now I'll have to go through and manually change it to an image instead of using elementor. What a pain

**Comment by vHeemstra** [comment]
> Hope Elementor will hurry up and get their CMS up-to-date! Even the paid Pro version still has Font Awesome version 5... 😒
> 
> Until then, for anyone looking for a way to import Font Awesome 6 icons as Custom Icon sets in Elementor Pro:
> 
> Have a look at [the guide I made here](https://gist.github.com/vHeemstra/39e747eede1d0c10d0350e189f5d7724).
> 
> Note: it is not the cleanest and/or most performant way to do it, but it works.
> 
> @amirhmoradi Feel free to use any of my code (if needed) for your plugin. If it is ready, will you let us know where to get it?

@Paul-Morris This might save you time 👆

**Comment by Paul-Morris** [comment]
> > Hope Elementor will hurry up and get their CMS up-to-date! Even the paid Pro version still has Font Awesome version 5... 😒
> > Until then, for anyone looking for a way to import Font Awesome 6 icons as Custom Icon sets in Elementor Pro:
> > Have a look at [the guide I made here](https://gist.github.com/vHeemstra/39e747eede1d0c10d0350e189f5d7724).
> > Note: it is not the cleanest and/or most performant way to do it, but it works.
> > @amirhmoradi Feel free to use any of my code (if needed) for your plugin. If it is ready, will you let us know where to get it?
> 
> @Paul-Morris This might save you time 👆

Thanks I appreciate it

**Comment by Nuyonuyonoina** [comment]
Incredible that hadn't been implementad yet. Clients are complaining for X icon...

**Comment by leecollings-sb** [comment]
So an issue at the moment for me is that I'm using the new Twitter X icon in my footer, however whenever I view a page that is using an icon element from Elementor, this force loads the old FA5 to load, and results in an ugly X in a box icon showing where my Twitter icon should be.

So is there any idea when Elementor will release a version that includes FA6? As others have pointed out, it's not exactly new anymore.

For now, I'll take a look at the guide that @vHeemstra made. For a corporate website, this looks awful.

**Comment by Paul-Morris** [comment]
> So an issue at the moment for me is that I'm using the new Twitter X icon in my footer, however whenever I view a page that is using an icon element from Elementor, this force loads the old FA5 to load, and results in an ugly X in a box icon showing where my Twitter icon should be.
> 
> So is there any idea when Elementor will release a version that includes FA6? As others have pointed out, it's not exactly new anymore.
> 
> ~For now, I'll take a look at the guide that @vHeemstra made.~ Can't do this, I don't have Elementor Pro.
> 
> For a corporate website, this looks awful.

As a quick temporary solution I ended up downloading Twitters official brand assets and using an svg image instead of an icon. You can select "Upload SVG" instead of "Icon library" when you hover over the Icon widget

**Comment by vHeemstra** [comment]
> As a quick temporary solution I ended up downloading Twitters official brand assets and using an svg image instead of an icon. You can select "Upload SVG" instead of "Icon library" when you hover over the Icon widget

@leecollings-sb FontAwesome also let's you download their free icons as SVG from the pop-op modal when you click on them on the Search Icons page: https://fontawesome.com/search?q=twitter&o=r&m=free

**Comment by leecollings-sb** [comment]
Thanks for the replies, but the issue that I'm having is actually not within Elementor, but that Elementor is causing the issue for the rest of my website.

In my theme footer I'm using FA6 library in my theme, and showing social icons (including the X icon). However, when I View a page that has any of Elementor's widgets that uses icons, that loads up the FA5 version in Elementor and stops my theme FA6 from loading, resulting in the X icon in the footer showing as an ugly X in a box (or image not found).

I can't think of a way to actually allow my theme icons to work properly because Elementor won't let me (because it's still using FA5 not 6).

Anyone know what I can do to get around this?

**Comment by vHeemstra** [comment]
> Thanks for the replies, but the issue that I'm having is actually not within Elementor, but that Elementor is causing the issue for the rest of my website.
> 
> In my theme footer I'm using FA6 library in my theme, and showing social icons (including the X icon). However, when I View a page that has any of Elementor's widgets that uses icons, that loads up the FA5 version in Elementor and stops my theme FA6 from loading, resulting in the X icon in the footer showing as an ugly X in a box (or image not found).
> 
> I can't think of a way to actually allow my theme icons to work properly because Elementor won't let me (because it's still using FA5 not 6).
> 
> Anyone know what I can do to get around this?

See if you can recompile/build FA6 with a custom main class name (like `.fa6` instead of `.fa`). This way, the FA5 `.fa` class will not interfere.

**Comment by toineenzo** [comment]
Lol, already more than 1 month without any commits and changes to the FA6 PR.... #24315
Kinda getting fed up. How do they want to release this the next update? Probably will not even be in 3.21, but probably in 4.0 or something.

**Comment by Byte-Developments** [comment]
Oh this would be awesome tbh.

**Comment by toineenzo** [comment]
Answer from Elementor:

Hi @toineenzo I'm sorry to inform you that this update ran into numerous issues, supporting FA6 without breaking existing sites that use FA5 is much more complicated than it might seem, we are still looking for ways to promote this update but I don't have a clear ETA at the moment,
Please note that the 'twitter/x' icon and the 'Thread' icon are already available since version 3.20

:(

**Comment by EdwardBlack** [comment]
> Hi toineenzo I'm sorry to inform you that this update ran into numerous issues, supporting FA6 without breaking existing sites that use FA5 is much more complicated than it might seem, we are still looking for ways to promote this update but I don't have a clear ETA at the moment, Please note that the 'twitter/x' icon and the 'Thread' icon are already available since version 3.20

No Worries Elementor. I loved you in the past, but _brick by brick_ I lost feelings for you. It's not me, it's you.
It's over - I have a new one. Not only that, but it's actually already going on for months, but I didn't want to tell you.
/unsubscribe

**Comment by masseym** [comment]
Why don’t you just add an option in settings allowing us to choose 5 or 6?  Then people can stay at 5 if it breaks things. Not rocket science

Matt Massey
MasseyMedia, Inc.

Phone  (936-225-4705
Mobile  (936) 225-2963
Email   ***@***.******@***.***>

On Aug 25, 2024, at 1:59 AM, Edward Black ***@***.***> wrote:

﻿

Hi toineenzo I'm sorry to inform you that this update ran into numerous issues, supporting FA6 without breaking existing sites that use FA5 is much more complicated than it might seem, we are still looking for ways to promote this update but I don't have a clear ETA at the moment, Please note that the 'twitter/x' icon and the 'Thread' icon are already available since version 3.20

No Worries Elementor. I loved you in the past, but brick by brick I lost feelings for you. It's not me, it's you.
It's over - I have a new one. Not only that, but it's actually already going on for months, but I didn't want to tell you.
/unsubscribe

**Comment by krlabs** [comment]
Need FA6 icons!

**Comment by EdwardBlack** [comment]
I'm out. Switched to BricksBuilder with my Agency and after 10 builds I am more than happy. Good luck to all Elementor users getting this fixed. #unsubscribed

**Comment by gerardreches** [comment]
+1

**Comment by Ricarditu** [comment]
Only after buying the Pro version I noticed theres is no FA6 direct integration in Elementor. And checking that there's more than 2 years to get this implemented... How is it possible?!
C'mon Elementor! Bring FA6!

**Comment by keyvanestermann** [comment]
For the people stumbling upon this issue in 2025, I got tired of waiting and made a Wordpress plugin :
https://github.com/keyvanestermann/better-fontawesome-wp

For now, it includes FontAwesome 6.7.2
It also changes the icons you can pick in the Elementor Icon Library 

Please note that this is a hack using an undocumented PHP Hook from Elementor, and it might break with any update
(currently works with Elementor 3.28.3)

**Comment by lphaweb** [comment]
@keyvanestermann does your plugin work with FontAwesom 7 pro ? :)

Elementor, you're two major release late of FA, that's a real shame.

**Comment by tobias992** [comment]
Yes, it's a big shame about this company. But hey, v4 is coming, and with it, a new wave of bugs.

None of our customers support this company anymore. They're all using the free GPL versions. In my opinion, everyone should do the same until they learn to listen to the community.

**Comment by gerardreches** [comment]
> But hey, v4 is coming, and with it, a new wave of bugs.

<img src="https://media3.giphy.com/media/v1.Y2lkPWJkM2VhNTdlNWM2ZHVsdnU0NW5rbjNtd2lqbmljaW92bzRuMzV4cjNudjZ2aDJuZyZlcD12MV9naWZzX3NlYXJjaCZjdD1n/3oFzmkkwfOGlzZ0gxi/giphy.gif"/>

I tried editor V4 a few days ago and it took me less than 5 minutes to find multiple bugs and limitations.

I can't believe that we are getting Font Awesome 7 before Elementor even updates to Font Awesome 6 while it is their main icons library.

**Comment by nicholaszein** [comment]
Hi there,

### ✈️ This request is being Migrated to the new [Discussions](https://github.com/elementor/elementor/discussions/categories/feature-request) section.

Once we migrate it, we encourage you to cast your **⬆️ Upvote** on the first comment if you feel it should be considered for our Roadmap.

Don't worry, your comments, reactions, and participation here will be accounted for when considering this Feature Request for our Roadmap.

Kind regards
